python/src/scipp/compat/mantid.py

```diff
@@ -481,19 +481,29 @@ def convert_monitors_ws(ws, converter, **ignored):
     return monitors
 
 
-def convert_Workspace2D_to_data_array(ws, advanced_geometry=False, **ignored):
+def convert_Workspace2D_to_data_array(ws,
+                                      advanced_geometry=False,
+                                      as_mask_workspace=False,
+                                      **ignored):
     dim, unit = validate_and_get_unit(ws.getAxis(0).getUnit().unitID())
     spec_dim, spec_coord = init_spec_axis(ws)
 
     coords_labs_data = _convert_MatrixWorkspace_info(
         ws, advanced_geometry=advanced_geometry)
     _, data_unit = validate_and_get_unit(ws.YUnit(), allow_empty=True)
-    stddev2 = ws.extractE()
-    np.multiply(stddev2, stddev2, out=stddev2)  # much faster than np.power
-    coords_labs_data["data"] = sc.Variable([spec_dim, dim],
-                                           unit=data_unit,
-                                           values=ws.extractY(),
-                                           variances=stddev2)
+    if as_mask_workspace:
+        coords_labs_data["data"] = sc.Variable([spec_dim, dim],
+                                               unit=data_unit,
+                                               values=ws.extractY(),
+                                               dtype=sc.dtype.bool)
+        coords_labs_data["masks"] = {'mask': coords_labs_data["data"].copy()}
```

Comment:
I don't think it makes sense to have this as mask and data. Just make it the data, users can then do `data.masks['mask'] = mask.data`.